lib/search-suggestions/index.jsx

```diff
@@ -0,0 +1,53 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import SmallSearchIcon from '../icons/search-small';
+import TagIcon from '../icons/tag';
+
+export class SearchSuggestions extends Component {
+  static displayName = 'SearchSuggestions';
+
+  static propTypes = {
+    onSearch: PropTypes.func.isRequired,
+    query: PropTypes.string.isRequired,
+    tags: PropTypes.array.isRequired,
+  };
+
+  render() {
+    const { query, onSearch, tags } = this.props;
+    // const screenReaderLabel =
+    //   'Search ' + (isTagSelected ? 'notes with tag ' : '') + placeholder;
+
+    return (
+      <div className="search-suggestions">
+        <div className="search-suggestion-row" onClick={() => onSearch(query)}>
+          <SmallSearchIcon />
+          {query}
+        </div>
+        {tags
+          .filter(function(tag) {
+            return tag.id.includes(query);
+          })
+          .map(tag => (
+            <div
+              key={tag.id}
+              className="search-suggestion-row"
+              onClick={() => onSearch(`tag:${tag.id}`)}
```

Comment:
Does this imply we can only search for _either_ one tag at a time _or_ a query string?

We could consider appending it to the current search string instead…

```js
onClick={ () => onSearch( `${ query } tag:${ tag.id }` );
```

It would leave some ugliness in there, like maybe we add the same tag twice, in which case we could handle this in other ways (remove the tag from the suggestion list if it's already in the query; toggle its inclusion when it's clicked so that a second click would remove it; update the `search` reducer to eliminate duplicate clauses; etc…)

Just noting the behavior that I think this implies which I didn't expect.

Reply:
This... is a really good point and definitely a case we need to handle.

Reply:
Wait, no, this doesn't make sense because the suggestion will be a matching string of the existing query. So we're sort of saying "oh, were you trying to find notes with this tag?"

I need to do more testing with multiple query arguments, it's not super clear how (or whether) that is supposed to work...